tests/test_stability_funs.py

```diff
@@ -349,3 +351,403 @@ def test_magwell_print(capsys):
         + "\n"
     )
     assert out.out == corr_out
+
+
+@pytest.mark.unit
+def test_ballooning_geometry(tmpdir_factory):
+    """Test the geometry coefficients used for the adjoint-ballooning solver.
+
+    The same coefficients are used for local gyrokinetic solvers which would
+    be useful when we couple DESC with GX/GS2 etc.
+    Observation: The larger the force error, the worse the tests behave. For
```

Comment:
can you explain to me  what this is doing?

Reply:
Calculating these coefficients using two different methods and cross-comparing. The ones with _alt are alterantive methods used by other code (ex. GX).

Reply:
Should the two methods be equivalent? why are there differences btwn the two methods?

Reply:
one way seems to assume JxB = grad(p) exactly I guess?

Reply:
It's all a consequence of jxb-grap being not exactly 0.